### PR TITLE
refactor: toast maintains its own 'visible' state

### PR DESCRIPTION
**Pull request by karanbirsingh**
#### Description of changes

This PR shows an idea for how the toast component could maintain its own state. The unit tests need to be updated to work properly with the react reference.

The team is looking at alternatives for the toast in general, so leaving this up as a draft

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/tests/common/components/toast.test.tsx [hunk 1]
```diff
@@ -30,36 +27,27 @@ describe('ToastTest', () => {
...
-    test('when timeout ends, callback is called & render is null', () => {
-        const timeoutId = 1;
-        windowUtilsMock
-            .setup(m => m.setTimeout(itIsFunction, 2000))
-            .callback((func, _) => func())
-            .returns(() => timeoutId)
-            .verifiable(Times.once());
 
-        onTimeoutMock.setup(m => m()).verifiable(Times.once());
+        // expect(subject.state.toastVisible).toBeTruthy();
```
Comment (karanbirsingh) [inline src/tests/unit/tests/common/components/toast.test.tsx]: doesn't work, the setState calls don't work until the component is mounted

File: src/common/components/cards/card-kebab-menu-button.tsx [hunk 2]
```diff
@@ -39,12 +37,12 @@ export interface CardKebabMenuButtonProps {
...
 export class CardKebabMenuButton extends React.Component<CardKebabMenuButtonProps, CardKebabMenuButtonState> {
+    private toastRef: React.RefObject<Toast>;
     constructor(props: CardKebabMenuButtonProps) {
         super(props);
-
+        this.toastRef = React.createRef();
         this.state = {
             showNeedsSettingsContent: false,
-            showingCopyToast: false,
             toastText: '',
```
Comment (dbjorge) [inline src/common/components/cards/card-kebab-menu-button.tsx]: `toastText` here is obsolete now, right?

File: src/tests/unit/tests/common/components/cards/card-kebab-menu-button.test.tsx [hunk 3]
```diff
@@ -166,16 +166,11 @@ describe('CardKebabMenuButtonTest', () => {
 
         rendered.find(ActionButton).simulate('click', event);
 
-        expect(rendered.state().showingCopyToast).toBe(false);
```
Comment (dbjorge) [inline src/tests/unit/tests/common/components/cards/card-kebab-menu-button.test.tsx]: Would be good to replace the removed assertions with new test content verifying the expected `show(...)` calls happen

File: src/common/components/cards/card-kebab-menu-button.tsx (hunk 2 above)
Comment (karanbirsingh) [reply src/common/components/cards/card-kebab-menu-button.tsx]: yes, forgot to remove

**Pull request closed without merging**